src/libinjector/linux/methods/linux_read_file.c

```diff
@@ -0,0 +1,371 @@
+            }
+
+            info->regs->rip = injector->syscall_addr;
+
+            event = VMI_EVENT_RESPONSE_SET_REGISTERS;
+
+            break;
+        }
+        case STEP2: // open file handle
+        {
+            if ( is_syscall_error(info->regs->rax) )
+            {
+                fprintf(stderr, "mmap syscall failed\n");
+                return cleanup(drakvuf, info, true);
+            }
+            PRINT_DEBUG("memory address allocated using mmap: %lx\n", info->regs->rax);
+
+            // save it for future use
+            injector->virtual_memory_addr = info->regs->rax;
+
+            PRINT_DEBUG("Opening file descriptor\n");
+            setup_open_syscall(injector, info->regs);
+
+            info->regs->rip = injector->syscall_addr;
+
+            event = VMI_EVENT_RESPONSE_SET_REGISTERS;
+            break;
+        }
+        case STEP3: // verify fd and setups first read syscall
+        {
+            if ( is_syscall_error(info->regs->rax) )
+            {
+                fprintf(stderr, "Could not open file in guest\n");
+                return cleanup(drakvuf, info, true);
+            }
+
+            injector->fd = info->regs->rax;
+            PRINT_DEBUG("File descriptor: %ld\n", injector->fd);
+
+            if (!setup_read_syscall(injector, info->regs, FILE_BUF_SIZE))
+                PRINT_DEBUG("Failed to setup read syscall\n");
+
+            info->regs->rip = injector->syscall_addr;
+
+            event = VMI_EVENT_RESPONSE_SET_REGISTERS;
+            break;
+        }
+        case STEP4: // loop till all chunks are written and then close the fd
+        {
+            if ( is_syscall_error(info->regs->rax) )
+            {
+                fprintf(stderr, "Could not read chunk from guest\n");
+                return cleanup(drakvuf, info, true);
+            }
+            injector->buffer.len = info->regs->rax;
+
+            if (!injector->buffer.len)
+            {
+                PRINT_DEBUG("Read file successful\n");
+                PRINT_DEBUG("File size: (%ld)\n", injector->buffer.total_processed);
+
+                if (!setup_close_syscall(injector, info->regs))
+                {
+                    PRINT_DEBUG("Failed to setup close syscall\n");
+                    return cleanup(drakvuf, info, true);
+                }
+            }
+            else
+            {
+                PRINT_DEBUG("Chunk read successful (%ld)\n", injector->buffer.len);
+                injector->buffer.total_processed += injector->buffer.len;
+
+                if (!write_buffer_to_file(drakvuf, info, injector->buffer.len))
+                    return cleanup(drakvuf, info, true);
+
+                if (!setup_read_syscall(injector, info->regs, FILE_BUF_SIZE))
+                {
+                    PRINT_DEBUG("Failed to setup read syscall\n");
+                    return cleanup(drakvuf, info, true);
+                }
+
+                injector->step_override = true;
```

Comment:
maybe let's renamed this variable to `injector->increment_step`?

Reply:
Umm, this change would require changes in other methods also. I would suggest a new PR for this

Reply:
ok